src/utils/utils.js

```diff
@@ -226,6 +226,22 @@ export function getActiveTab(callback) {
 	});
 }
 
+/**
+ * Query the first tab that matches the url argument
+ * @memberOf BackgroundUtils
+ *
+ * @param {function} callback		function to call if tab found
+ * @param {string} url				the tab url to search for
+ */
+export function getTabByUrl(callback, url) {
```

Comment:
@wlycdgrfromflatiron please rewrite this function so that we check for `chrome.runtime.lastError` in the callback function so that we can be assured that `tabs` will be an array, and that we handle an array of size `0` to whatever length.

Reply:
I've rewritten getTabByUrl to invoke a fallback function argument in case of chrome.runtime.lastErrror or if the query returns an empty array, using the error handling code in getTab as a template.